pkg/client/aad.go

```diff
@@ -65,3 +61,27 @@ func (c *aadClientImpl) GetAadToken(ctx context.Context, clientID, clientSecret,
 
 	return authResult.AccessToken, nil
 }
+
+type tokenAcquirer interface {
+	Acquire(ctx context.Context, scopes []string) (confidential.AuthResult, error)
+}
+
+type confidentialTokenAcquirer struct {
+	confidential.Client
+}
+
+func newConfidentialTokenAcquirer(authority, clientID, clientSecret string) (tokenAcquirer, error) {
```

Comment:
it's probably fine if this logic isn't covered since we're pretty much _only_ calling out to an external lib with minimal error handling - no fancy business logic or anything